### PR TITLE
Add generic method for overriding specific config variables via env vars

### PR DESCRIPTION
**Pull request by sbc100**
For any config variable such as LLVM_ROOT you can now always set
EM_LLVM_ROOT to override it.

### CODE REVIEW

File: tests/test_sanity.py [hunk 1]
```diff
@@ -874,16 +874,6 @@ def prep():
...
-      config = open(CONFIG_FILE).read()
-      assert '''BINARYEN_ROOT = os.path.expanduser(os.getenv('BINARYEN', ''))''' in config, config # setup created it to be ''
-      print('created config:')
-      print(config)
-      restore_and_set_up()
-      config = open(CONFIG_FILE).read()
-      config = config.replace('BINARYEN_ROOT', '''BINARYEN_ROOT = os.path.expanduser(os.getenv('BINARYEN', '')) # ''')
-      print('modified config:')
-      print(config)
-      open(CONFIG_FILE, 'w').write(config)
```
Comment (kripken) [inline tests/test_sanity.py]: why was this test code removed?

Comment (sbc100) [reply tests/test_sanity.py]: As far as I can tell this wasn't doing anything.  It looks like it verifies that BINARYEN_ROOT defaults to empty string.  Then it sets BINARY_ROOT to the empty string.   I'm I missing something?

**Comment by sbc100** [comment]
The llvm env var was checked for by the default config.  Anyone with an existing config will continue to have that work.

However when they regenerate their config file it will stop working because we changed the default config.  I guess it depends if we want to say that the behaviour of the default config should be preserved.  Perhaps we could at least add a changelog entry and send out an email.

**Comment by sbc100** [comment]
OK I reverted most of the changes to the default config file.

File: tests/test_sanity.py (hunk 1 above)
Comment (kripken) [reply tests/test_sanity.py]: Hmm I think you're right, I can't see what this code does either.

File: ChangeLog.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ v1.38.48: 10/11/2019
    on the main thread. #9567
  - Remove `EMCONFIGURE_JS`. Since #6269 we have set it to "2" which means never
    use native, always use JS.
+ - Add support for override config variables using environment varaibles.  Any
```
Comment (kripken) [inline ChangeLog.md]: typo in `variables`

File: ChangeLog.md [hunk 3]
```diff
@@ -32,6 +32,8 @@ v1.38.48: 10/11/2019
...
+   config varaible `FOO` can be overridden by `EM_FOO` in the environment.
```
Comment (kripken) [inline ChangeLog.md]: typo in `variable`

File: tools/shared.py [hunk 4]
```diff
@@ -348,7 +351,11 @@ def parse_config_file():
 
   # Only popogate certain settings from the config file.
```
Comment (kripken) [inline tools/shared.py]: pre-existing typoe in `propagate`

Comment (kripken) [reply tools/shared.py]: heh, typo in my last comment in `typo` ;)

**Pull request merged**